src/node/internal/crypto_dh.ts

```diff
@@ -0,0 +1,203 @@
+  public getPrivateKey = dhGetPrivateKey;
+  public getPublicKey = dhGetPublicKey;
+
+  public setPublicKey(this: DiffieHellman, key: ArrayLike, encoding?: string): DiffieHellman {
+    key = getArrayBufferOrView(key, 'key', encoding);
+    this[kHandle].setPublicKey(key);
+    return this;
+  }
+
+  public setPrivateKey(this: DiffieHellman, key: ArrayLike, encoding?: string): DiffieHellman {
+    key = getArrayBufferOrView(key, 'key', encoding);
+    this[kHandle].setPrivateKey(key);
+    return this;
+  }
+
+  public get verifyError(): number { return this[kHandle].getVerifyError(); }
+}
+
+export class DiffieHellmanGroup {
+  private [kHandle]: cryptoImpl.DHHandle;
+
+  constructor(name: string) {
+    this[kHandle] = cryptoImpl.DHGroupHandle(name);
+  }
+
+  public computeSecret = dhComputeSecret;
+  public generateKeys = dhGenerateKeys;
+  public getPrime = dhGetPrime;
+  public getGenerator = dhGetGenerator;
+  public getPrivateKey = dhGetPrivateKey;
+  public getPublicKey = dhGetPublicKey;
+  public get verifyError(): number { return this[kHandle].getVerifyError(); }
+}
+
+function dhGenerateKeys(this: any, encoding?: string): Buffer|string {
+  const keys = this[kHandle].generateKeys();
+  return encode(keys, encoding);
+}
+
+function dhComputeSecret(this: any, key: ArrayLike, inEnc?: string,
+                         outEnc?: string): Buffer|string {
+  key = getArrayBufferOrView(key, 'key', inEnc);
+  const ret = this[kHandle].computeSecret(key);
+  if (typeof ret === 'string')
+    throw new ERR_CRYPTO_ECDH_INVALID_PUBLIC_KEY();
+  return encode(ret, outEnc);
+}
+
+function dhGetPrime(this: any, encoding?: string): Buffer|string {
+  const prime = this[kHandle].getPrime();
+  return encode(prime, encoding);
+}
+
+function dhGetGenerator(this: any, encoding?: string): Buffer|string {
+  const generator = this[kHandle].getGenerator();
+  return encode(generator, encoding);
+}
+
+function dhGetPublicKey(this: any, encoding?: string): Buffer|string {
+  const key = this[kHandle].getPublicKey();
+  return encode(key, encoding);
+}
+
+function dhGetPrivateKey(this: any, encoding?: string): Buffer|string {
+  const key = this[kHandle].getPrivateKey();
+  return encode(key, encoding);
+}
+
+function encode(buffer: ArrayBuffer, encoding?: string): Buffer|string {
+  if (encoding && encoding !== 'buffer')
+    return Buffer.from(buffer).toString(encoding);
+  return Buffer.from(buffer);
+}
+
+export function createDiffieHellman(sizeOrKey: number|ArrayLike,
+                                    keyEncoding?: number|string, generator?: number|ArrayLike,
+                                    genEncoding?: string): DiffieHellman {
+  return new DiffieHellman(sizeOrKey, keyEncoding, generator, genEncoding);
+}
+
+export function createDiffieHellmanGroup(name: string): DiffieHellmanGroup {
+  return new DiffieHellmanGroup(name);
+}
+
+export function getDiffieHellman(name: string): DiffieHellmanGroup {
+  return createDiffieHellmanGroup(name);
+}
```

Comment:
Nit: empty new lines at the end of files...

```suggestion
}

```